src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -544,43 +545,16 @@ protected override void CreateIndicesForNonTypeMembers(ITypeDefinition typeDef)
                 this.AddDefIfNecessary(_methodDefs, methodDef);
                 var methodChange = _changes.GetChange(methodDef);
 
-                if (methodChange == SymbolChange.Added)
-                {
-                    _firstParamRowMap.Add(GetMethodDefinitionHandle(methodDef), _parameterDefs.NextRowId);
-                    foreach (var paramDef in this.GetParametersToEmit(methodDef))
-                    {
-                        _parameterDefs.Add(paramDef);
-                        _parameterDefList.Add(paramDef, methodDef);
-                    }
-                }
-                else if (methodChange == SymbolChange.Updated)
-                {
-                    // If we're re-emitting parameters for an existing method we need to find their original row numbers
-                    // and reuse them so the EnCLog, EnCMap and CustomAttributes tables refer to the right rows
+                EmitMethod(methodDef, methodChange);
+            }
 
-                    // Unfortunately we have to check the original metadata and deltas separately as nothing tracks the aggregate data
-                    // in a way that we can use
-                    var handle = GetMethodDefinitionHandle(methodDef);
-                    if (_previousGeneration.OriginalMetadata.MetadataReader.GetTableRowCount(TableIndex.MethodDef) >= MetadataTokens.GetRowNumber(handle))
-                    {
-                        EmitParametersFromOriginalMetadata(methodDef, handle);
-                    }
-                    else
-                    {
-                        EmitParametersFromDelta(methodDef, handle);
-                    }
-                }
+            foreach (var methodDef in _changes.GetDeletedMethods(typeDef))
+            {
+                var oldMethodDef = (IMethodDefinition)methodDef.GetCciAdapter();
+                var newMethodDef = new DeletedMethodDefinition(oldMethodDef, methodDef);
```

Comment:
>oldMethodDef [](http://example.com/codeflow?start=63&length=12)

Seems like this argument is redundant since the oldMethodDef can be retrieved from methodDef.